pkg/image/registry/etcd/etcd.go

```diff
@@ -82,10 +84,39 @@ func (r *Etcd) CreateImage(ctx kapi.Context, image *api.Image) error {
 		return err
 	}
 
-	err = r.CreateObj(key, image, 0)
+	err = r.AtomicUpdate(key, &api.Image{}, func(obj runtime.Object) (runtime.Object, error) {
+		existing, ok := obj.(*api.Image)
+		if !ok {
+			return nil, fmt.Errorf("unexpected object: %#v", obj)
+		}
+		if isNewObject(existing.ResourceVersion) {
+			return image, nil
+		}
+		if equivalentImage(existing, image) {
+			return existing, nil
+		}
+		return nil, apierrs.NewAlreadyExists("image", image.Name)
+	})
 	return etcderr.InterpretCreateError(err, "image", image.Name)
 }
 
+// isNewObject returns true if the provided resource version indicates the object has not been previously persisted.
+func isNewObject(resourceVersion string) bool {
```

Comment:
It should go upstream, didn't want to do it here.

----- Original Message -----
> >  	return etcderr.InterpretCreateError(err, "image", image.Name)
> >  }
> >  
> > +// isNewObject returns true if the provided resource version indicates the
> > object has not been previously persisted.
> > +func isNewObject(resourceVersion string) bool {
> 
> Shouldn't this be a public method in some helper package somewhere? I assume
> this will be needed by other packages as well.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/635/files#r23165708